### PR TITLE
fix: Amavis should reload config for `/etc/postfix/vhost` updates

### PR DESCRIPTION
**Pull request by polarathene**
# Description

When `/etc/postfix/vhost` is updated from change detection, Amavis did not refresh that source, and thus would not be aware of any new domains, ignoring them for spam/virus checking.

A test could probably be added, my time is focused on prepping PRs with the time I have available right now.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: target/scripts/check-for-changes.sh [hunk 1]
```diff
@@ -86,6 +88,16 @@ function _get_changed_files
   grep -Fxvf "${CHKSUM_CURRENT}" "${CHKSUM_NEW}" | sed -r 's/^\S+[[:space:]]+//'
 }
 
+function _reload_amavis
+{
+  if [[ ${CHANGED} =~ ${DMS_DIR}/postfix-accounts.cf ]] || [[ ${CHANGED} =~ ${DMS_DIR}/postfix-virtual.cf ]]
+  then
+    # /etc/postfix/vhost was updated, amavis must refresh it's config by
+    # reading this file again in case of new domains, otherwise they will be ignored.
+    amavisd-new reload
```
Comment (polarathene) [inline target/scripts/check-for-changes.sh:97]: Presumably `supervisorctl restart amavis` should be used.

I've [not received any feedback about better understanding why `postfix reload` was an issue in the past](https://github.com/docker-mailserver/docker-mailserver/pull/2624#issuecomment-1148043165) (_may have been due to the present `postfix-wrapper.sh` script?_).

Comment (georglauterbach) [reply target/scripts/check-for-changes.sh:97]: Truth be told, I don‘t know 🙈 @casperklein do you have an idea?:)

File: target/scripts/check-for-changes.sh [hunk 2]
```diff
@@ -59,6 +59,8 @@ function _check_for_changes
     # While some config changes may be properly applied by Postfix or Dovecot
     # via their 'reload' commands; some may require restarting?:
     _log_with_date 'debug' 'Restarting services due to detected changes'
+
+    _reload_amavis
```
Comment (polarathene) [inline target/scripts/check-for-changes.sh:63]: Only do so if Amavis service is active:

```suggestion
    [[ ${ENABLE_AMAVIS} -eq 1 ]] && _reload_amavis
```

File: target/scripts/check-for-changes.sh (hunk 1 above)
Comment (casperklein) [reply target/scripts/check-for-changes.sh:97]: The usage of `amavisd-new reload` should be fine. However, if for some reason a complete service restart is needed, `supervisorctl restart amavis` should be used.

**Review by casperklein — APPROVED** [review]

**Review by georglauterbach — APPROVED** [review]

**Pull request merged**